diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/SingleConstructorRule.kt

```diff
@@ -157,14 +176,38 @@ class SingleConstructorRule(configRules: List<RulesConfig>) : DiktatRule(
             }
         }
 
+        val initBody: MutableList<String> = mutableListOf()
+
+        initBody.addAll(otherStatements.map { it.text })
+        initBody.addAll(nonTrivialAssignments.keys.map { it.text })
+
+        val argumentListOfSecondaryCtor: List<KtParameter> = (secondaryCtor.psi as KtSecondaryConstructor)
+            .valueParameters
+            .filter { arg -> arg.name !in nonTrivialSecondaryCtorParameters.map { it.name } }
+            .filter { arg -> arg.name !in declarationsAssignedInCtor.map { it.name } }
+            .filter { arg -> initBody.any { expr -> arg.name.toString() in expr } }
+
+        comments?.forEach { (comment, nextExpression) ->
+            if (initBody.indexOf(nextExpression?.text) != -1) {
+                initBody.add(initBody.indexOf(nextExpression?.text), comment)
+            }
+        }
+
         if (otherStatements.isNotEmpty() || nonTrivialAssignments.isNotEmpty()) {
             findChildByType(CLASS_BODY)?.run {
-                val classInitializer = kotlinParser.createNode(
-                    """|init {
-                       |    ${(otherStatements + nonTrivialAssignments.keys).joinToString("\n") { it.text }}
+                val classInitializer = if (argumentListOfSecondaryCtor.isEmpty()) {
+                    kotlinParser.createNodeForInit(
+                        """|init {
+                           |    ${initBody.joinToString("\n")}
+                           |}
+                        """.trimMargin())
+                } else {
+                    kotlinParser.createNodeForSecondaryConstructor(
```

Comment:
We are trying to replace the secondary constructor with a primary one. Maybe it would be better not to change anything if the case is unfixable, rather than replace one secondary constructor with a new one? Or what are trying to do here?

Reply:
What if we can remove some parameters and simplify secondary constructor?

Reply:
Then it should be decorated differently. Now, if the rule detects redundant parameters in `check` mode, user will see something like `If there is a single constructor, it should be converted to primary`. And fix logic will leave it there, but remove some parameters. If we will add such logic, it should have a separate warning in `Warnings.kt`. Let's leave it for #1124.

Reply:
Makes sense, agree